src/core/node/object-node.ts

```diff
@@ -216,7 +216,7 @@ export class ObjectNode<C, S, T> extends BaseNode<C, S, T> {
     const type = this.type
 
     try {
-      this.storedValue = type.createNewInstance(this._childNodes)
+      this.storedValue = type.createNewInstance(this._childNodes) as typeof this.storedValue
```

Comment:
`createNewInstance` returns a `T`, which is correct, but we're gonna turn `T` into an `IStateTreeNode` by setting some special properties below, so it definitely _will_ be an `IStateTreeNode` eventually.

If avoiding this `as T` is desirable, I would rewrite the code in this way:

```
const instance = type.createNewInstance(this._childNodes);
this.storedValue = this.preboot(instance);
type.finalizeNewInstance(this, this.storedValue)
```

Because `preboot` adds the special sauce that makes `instance` an STN.

Reply:
I think this is sufficient. I'm going to add a comment pointing to this PR about it for posterity. Thank you!